source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,424 @@
+            "city" : "Hawthorne",
+            "state" : "CA",
+            "zipcode" : "90250"
+         },
+         "geo" : {
+            "type" : "Point",
+            "coordinates" : [
+               -118.36559,
+               33.897167
+            ]
+         }
+      }
+   }
+
+The following example creates a ``2dsphere`` index on the ``location.geo`` field:
+
+.. important::
+
+   Attempting to create a geospatial index on a field that is already covered by a geospatial index results in an error.
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-geospatial-index
+   :end-before: end-geospatial-index
+   :dedent:
+
+The following is an example of a geospatial query using the "location.geo" index:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-geospatial-query
+   :end-before: end-geospatial-query
+   :dedent:
+
+MongoDB also supports ``2d`` indexes for calculating distances on a Euclidean plane and
+for working with the "legacy coordinate pairs" syntax used in MongoDB 2.2 and earlier.
+To learn more, see :manual:`Geospatial Queries </geospatial-queries>` in the Server manual.
+
+Unique Indexes
+~~~~~~~~~~~~~~
+
+**Unique indexes**  ensure that the indexed fields do not store duplicate values. By
+default, MongoDB creates a unique index on the ``_id`` field during the creation of a
+collection. To create a unique index, specify the fields that you want to prevent
+duplication on and set the ``Unique`` option to ``true``.
+
+The following example creates a unique, descending index on the ``theaterId`` field within
+the ``sample_mflix.theaters`` collection.
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-unique-index
+   :end-before: end-unique-index
+   :dedent:
+
+If you attempt to perform a write operation that stores a duplicate value
+that violates the unique index, MongoDB will throw an error that resembles
+the following:
+
+.. code-block:: none
+   :copyable: false
+
+   E11000 duplicate key error index
+
+To learn more, see :manual:`Unique Indexes </core/index-unique>` in the Server manual.
+
+Wildcard Indexes
+~~~~~~~~~~~~~~~~
+
+**Wildcard indexes** enable queries against unknown or arbitrary fields.
+These indexes can be beneficial if you are using a dynamic schema.
+
+The following example creates an ascending wildcard index on all
+values of the ``location`` field within the ``sample_mflix.theaters`` collection,
+including values nested in subdocuments and arrays:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-wildcard-index
+   :end-before: end-wildcard-index
+   :dedent:
+
+For more information, see the :manual:`Wildcard Indexes</core/index-wildcard>` page in the
+Server manual.
+
+.. _search-indexes:
+
+Atlas Search Indexes
```

Comment:
Given the confusion between `text` and Atlas Search indexes, I would suggest moving the two side-by-side for easier documentation and comparison. If you're following a standard cross-driver ordering, then it's fine to leave as-is.